eventlog2speedscope/src/Speedscope/Main.hs

```diff
@@ -127,22 +138,31 @@ createTables conn =
     "CREATE TABLE IF NOT EXISTS\
     \ events (at INTEGER, thread INTEGER, type TEXT, frame INTEGER)"
 
-newtype FrameFilter = FrameFilter (Set FrameId)
+newtype FrameFilter = FrameFilter (ThreadAnalysis -> (Set FrameId))
```

Comment:
I would have expected `ThreadAnalysis -> Bool` here, or a record type with two `Set`s. How does this work?

Reply:
I refactored the implementation, I hope it's clearer now.

My understanding is that a `FrameId` has an associated `ThreadAnalysis` (stack). When filtering with `--not-matching-children`, we look at the stack and make sure that none of the ids in the stack appear in the `notMatchingChildren` set. Does that make sense or am I misunderstanding something?